code/datums/craft/recipe.dm

```diff
@@ -67,6 +67,52 @@
 	var/datum/craft_step/CS = steps[step]
 	return CS.apply(I, user, target, src)
 
+/datum/craft_recipe/proc/build_batch(mob/living/user, amount)
```

Comment:
make this proc utilize try_build() instead of copypasting code

Reply:
That would result in lots of stacks containing only one item spawning when the result is a stack. This proc puts all items in one stack in that case.

Reply:
make try_build() return result if succeeded and stack all stackables in build_batch()

Reply:
And where should the spawn code from try_build go?

Reply:
it stays there, you just merge all stackables that try_build() returned

Reply:
resolved in 54bced1e